classes/core/Controllers/Middlewares/FillContextBasedOnUri.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PKP\core\Controllers\Middlewares;
+
+use APP\core\Application;
+use Closure;
+use Illuminate\Http\Request;
+use PKP\core\Core;
+use PKP\db\DAORegistry;
+
+class FillContextBasedOnUri
+{
+    /**
+     * Handle an incoming request.
+     */
+    public function handle(Request $request, Closure $next)
```

Comment:
This duplicates code from PKPRouter::getContext, I think; it would be better to call that directly if possible. (A lot of that code is old and ready for removal since we're no longer supporting apps with different context depths -- OCS and OHS used to do this -- so spreading it out further into the codebase is code rot.)

Reply:
I tried to keep the logic on this middleware simple. I got the logic to retrieve context based on path data, did the database requests and inject it as an attribute on Request itself.

The `PKPRouter::getContext` isn't a static method, so I couldn't use it easier here. I will always need to inject and construct a new PKPRouter instance, to call the `getContext`. Thus, keeping it extremely coupled. :(